spec/abci++/abci++_methods_002_draft.md

```diff
@@ -318,8 +318,14 @@ title: Methods
             * If the Application wants to add a new transaction, then the Application should include it in `tx_records` and _mark_ it as `ADD`. In this case, Tendermint will add it to the mempool.
         * The Application should be aware that removing and adding transactions may compromise _traceability_.
           > Consider the following example: the Application transforms a client-submitted transaction `t1` into a second transaction `t2`, i.e., the Application asks Tendermint to remove `t1` and add `t2` to the mempool. If a client wants to eventually check what happened to `t1`, it will discover that `t_1` is not in the mempool or in a committed block, getting the wrong idea that `t_1` did not make it into a block. Note that `t_2` _will be_ in a committed block, but unless the Application tracks this information, no component will be aware of it. Thus, if the Application wants traceability, it is its responsability to support it. For instance, the Application could attach to a transformed transaction a list with the hashes of the transactions it derives from. 
-        * If the Application modifies the set of transactions, the modified transactions MUST NOT exceed the configured maximum size `RequestPrepareProposal.max_tx_bytes`.
     * If the Application does not modify the preliminary set of transactions `txs`, then it sets `ResponsePrepareProposal.modified_tx_status` to `UNMODIFIED`. In this case, Tendermint will ignore the contents of `ResponsePrepareProposal.tx_records`.
+    * Tendermint MAY include a list of transactions in `RequestPrepareProposal.txs` whose total size in bytes exceeds `RequestPrepareProposal.max_tx_bytes`.
+      Therefore:
+        * When setting `ResponsePrepareProposal.modified_tx_status` to `MODIFIED`, the Application MUST always make sure that
+          the `RequestPrepareProposal.max_tx_bytes` limit is respected by the list of transaction records it returns
+          in `ResponsePrepareProposal.tx_records`, excluding those marked as `REMOVE`.
+        * If the size of `RequestPrepareProposal.txs` is greater than `RequestPrepareProposal.max_tx_bytes`, it is _invalid_ to return
+          `ResponsePrepareProposal.modified_tx_status` set to `UNMODIFIED`: Tendermint will panic.
```

Comment:
Just a suggestion:

* Tendermint MAY include a list of transactions in `RequestPrepareProposal.txs` whose total size in bytes exceeds `RequestPrepareProposal.max_tx_bytes`. Therefore, if the size of `RequestPrepareProposal.txs` is greater than `RequestPrepareProposal.max_tx_bytes`:
  * the Application MUST make sure that the `RequestPrepareProposal.max_tx_bytes` limit is respected by the list of transaction records it returns in `ResponsePrepareProposal.tx_records`, excluding those marked as `REMOVE`.
  * the Application MUST set `ResponsePrepareProposal.modified_tx_status` to `MODIFIED`: Tendermint will panic otherwise.